### ISSUE TITLE
Include binaries as fallback if auto-tools is missing

### ISSUE THREAD

**Issue by jonasfj** [paixaop/node-sodium#37]
This springs out of the [discussion surrounding the develop branch](https://github.com/paixaop/node-sodium/pull/27#issuecomment-76093352).

We can embed prebuilt binaries for libsodium. Then make a custom `npm install` script tries to run `make sodium` and if that fails prints a big warning and falls back to pre-built binaries.
This is a common pattern on NPM. 

The libsodium project already provides binaries for Windows:
https://download.libsodium.org/libsodium/releases/

We can easily include a linux binary for x64, and x86... We can either find a trick to link against an old glibc ABI, or we can use a docker image (or vagrant box) with debian sqeeze to build then it'll certainly have an old version of glibc.

Notes,
 * glibc provides backward compatibility. So to ensure maximum compatibility, we should build against a really old version of glibc. All the other libs can be linked statically.
 * We need to check licensing of linked libraries before we can distribute binaries as an element of this project.

Is this something we should explore?

**Comment by jonasfj** [comment]
Correction it's a custom `preinstall` script we need to make... And we should assume `node-gyp` is in `PATH`..

Or we have to do it in the `.gyp` file... and basically replace the libsodium build process.
http://n8.io/converting-a-c-library-to-gyp/

Apparently install script is an anti-pattern :)
https://docs.npmjs.com/misc/scripts#note-install-scripts-are-an-antipattern

**Comment by paixaop** [comment]
Indeed the gyp file compiles lib sodium without using the library's original build system.

The issue that we've run into in the past is that by not running configure on libsodium the platform/architecture specific defines are not created which will cause incompatibilities and compilation errors.

We could create different conditional sections in the GYP for the supported platforms but then we would need to add and maintain each platform's section. This is not very maintainable. 
Maybe the mid ground is to have a section for windows which would be the only system where automake tools are not easily available.

**Comment by jonasfj** [comment]
hmm... maintain a gyp section for windows would probably be the solution. Better than shipping binaries :)

Closing this issue, as a single gyp section is better for windows, than binaries... Also easier to maintain, binaries are the worst :)

Side note, I tested the develop branch locally and on heroku yesterday and it works both places. Granted We still need the ./configure hooked up to npm install, before the `develop` branch is usable. But that should be a simple matter...
